### ISSUE TITLE
Rule Change: Improve `max-lines-per-function` on test files

### ISSUE THREAD

**Issue by moshest** [eslint/eslint#15981]
### What rule do you want to change?

max-lines-per-function

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
describe('my test', () => {
  describe('section 1', () => {
    it('should test1', () => {
      // ...
    });
  
    it('should test2', () => {
      // ...
    });
  });
});
```

### What does the rule currently do for this code?

count lines for each test function `it` as well as for the global section function `describe`.

### What will the rule do after it's changed?

Ignore functions given for `describe` function and only check functions under `it`

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
I’m sorry, core rules cannot know about the test runner being used, so we can’t implement this change. You can always create a custom rule by copying the existing rule and modifying it to do what you want.

**Comment by moshest** [comment]
So maybe we can address that with additional options?

```json
{
  "ignoreTestFunctions": ["describe"]
}
```

**Comment by moshest** [comment]
I submit a new issue with different suggestion here: https://github.com/eslint/eslint/issues/16000
